src/re/Types.re

```diff
@@ -0,0 +1,93 @@
+[@bs.deriving jsConverter]
+type id = [
+  | [@bs.as "Skin"] `Skin
+  | [@bs.as "Hair"] `Hair
+  | [@bs.as "FacialHair"] `FacialHair
+  | [@bs.as "Body"] `Body
+  | [@bs.as "Eyes"] `Eyes
+  | [@bs.as "Head"] `Head
+  | [@bs.as "Mouth"] `Mouth
+  | [@bs.as "Nose"] `Nose
+  | [@bs.as "Background"] `Background
+];
+
+type color = [
+  | `SkinColor
+  | `HairColor
+  | `FacialHairColor
+  | `BodyColor
+  | `BgColor
+];
+
+type style = [
+  | `HairStyle
+  | `SkinStyle
+  | `FacialHairStyle
+  | `BodyStyle
+  | `EyesStyle
+  | `MouthStyle
+  | `NoseStyle
+];
+
+type key = [
+  | `SkinColor
+  | `HairColor
+  | `FacialHairColor
+  | `BodyColor
+  | `BgColor
+  | `HairStyle
+  | `SkinStyle
+  | `FacialHairStyle
+  | `BodyStyle
+  | `EyesStyle
+  | `MouthStyle
+  | `NoseStyle
+];
```

Comment:
Try to use a union type here: https://reasonml.org/blog/union-types-in-bucklescript

Reply:
Also https://dev.to/yawaramin/emulating-typescript-union-types-with-reasonml-3ea0